torch/csrc/autograd/variable.h

```diff
@@ -145,6 +145,10 @@ namespace impl {
   /// Variable.
   TORCH_API void rebase_history(const Variable&, Edge gradient_edge);
 
+  /// Mark a view such that its history cannot be rebased.
+  /// This is only valid to call on views.
+  TORCH_API void disable_rebase_history(Variable&);
```

Comment:
Is this something that morally is only called at view construction time, or can a Variable have its rebase history disabled at some later point in time? When would I call this function?

Reply:
If it is only called at construction time, can it be moved into the constructor directly?

Reply:
After offline discussion. No. So this is moved to the constructor. And the other functions are cleaned-up to reduce the number of boolean arguments.